File: argv.py
#
# -*- coding: utf-8 -*-
# $Date: 2021/10/08 00:17:54 $
#
import sys



print('sys.argv             : ', sys.argv)
print('type(sys.argv)       : ', type(sys.argv))
print()

print('sys.argv[0])         : ', sys.argv[0])
print('type(sys.argv[0])    : ', type(sys.argv[0]))
print()

print('len(sys.argv)        : ', len(sys.argv))
print('type(len(sys.argv))  : ', type(len(sys.argv)))
print()



def type_file(file):
	print(file)
	i = 0
	fh = open(file)
	line = fh.readline()

	while line:
		i += 1
		print(i, "\t:", line, end='')
		line = fh.readline()

	fh.close()
	print()



argv = sys.argv
argc = len(argv)

for i in range(argc):
	type_file(argv[i])

quit()

File: argv.c
//
// -*- coding: utf-8 -*-
// $Date: 2021/10/13 22:43:05 $
//

#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <stdint.h>
#include <unistd.h>

int main(int argc, char *argv[])
{
	char buff[256];
	getcwd(buff, 256);
	printf("cd %s\n", buff);
	strcat(buff, "\\");

	for(int i = 0; i < argc; i++){
		printf("%2d %s\n", i, argv[i]);
		/*
		if(strncmp(buff, argv[i], strlen(buff)) == 0){
			printf("  => %s\n", argv[i]+strlen(buff));
		}else{
			;;;
		}
		*/
	}
}
